snuba/datasets/discover.py

```diff
@@ -0,0 +1,236 @@
+        depending on the detected dataset
+        """
+        return None
+
+    def get_columns(self) -> ColumnSet:
+        common = ColumnSet([
+            ('event_id', FixedString(32)),
+            ('project_id', UInt(64)),
+            ('timestamp', DateTime()),
+            ('deleted', UInt(8)),
+            ('retention_days', UInt(16)),
+            ('platform', Nullable(String())),
+            ('sentry:release', Nullable(String())),
+            ('sentry:dist', Nullable(String())),
+            ('sentry:user', Nullable(String())),
+            # User
+            ('user_id', Nullable(String())),
+            ('username', Nullable(String())),
+            ('email', Nullable(String())),
+            ('ip_address', Nullable(String())),
+        ])
+
+        return common + self.get_events_only_columns() + self.get_transactions_only_columns()
+
+    def get_events_only_columns(self) -> ColumnSet:
+        return ColumnSet([
+            ('group_id', Nullable(UInt(64))),
+            ('primary_hash', Nullable(FixedString(32))),
+            ('type', Nullable(String())),
+            # Promoted tags
+            ('level', Nullable(String())),
+            ('logger', Nullable(String())),
+            ('server_name', Nullable(String())),
+            ('transaction', Nullable(String())),
+            ('site', Nullable(String())),
+            ('url', Nullable(String())),
+
+            ('message', Nullable(String())),
+            ('search_message', Nullable(String())),
+            ('title', Nullable(String())),
+            ('location', Nullable(String())),
+            ('culprit', Nullable(String())),
+            ('received', Nullable(DateTime())),
+            ('geo_country_code', Nullable(String())),
+            ('geo_region', Nullable(String())),
+            ('geo_city', Nullable(String())),
+            ('sdk_name', Nullable(String())),
+            ('sdk_integrations', Nullable(Array(String()))),
+            ('sdk_version', Nullable(String())),
+            ('version', Nullable(String())),
+            ('http_method', Nullable(String())),
+            ('http_referer', Nullable(String())),
+            ('exception_stacks.type', Nullable(String())),
+            ('exception_stacks.value', Nullable(String())),
+            ('exception_stacks.mechanism_type', Nullable(String())),
+            ('exception_stacks.mechanism_handled', Nullable(UInt(8))),
+            ('exception_frames.abs_path', Nullable(String())),
+            ('exception_frames.filename', Nullable(String())),
+            ('exception_frames.package', Nullable(String())),
+            ('exception_frames.module', Nullable(String())),
+            ('exception_frames.function', Nullable(String())),
+            ('exception_frames.in_app', Nullable(UInt(8))),
+            ('exception_frames.colno', Nullable(UInt(32))),
+            ('exception_frames.lineno', Nullable(UInt(32))),
+            ('exception_frames.stack_level', Nullable(UInt(16))),
```

Comment:
This is a nested column on events. Is this intentional ?

Reply:
Yeah, i think it's easier if the Discover dataset doesn't need to know about columns that might be nested in one of the datasets. It's going to be a lot easier to map and fill in values and types if columns are not nested. More immediately `ColumnSet.get()` won't work when we pass the expanded column name on nested types.